### ISSUE TITLE
Exception of type 'Couchbase.Lite.HttpResponseException' was thrown.   at Couchbase.Lite.Util.TransientErrorRetryHandler.HandleTransientErrors 

### ISSUE THREAD

**Issue by ykurniawan** [couchbase/couchbase-lite-net#849]
Hi,

I'm using CBL 1.3.1 and I got so many crash logs with following exception. Any idea or place where I can catch the exception? I could not catch the exception from replication changed event. Thank you.

Application Specific Information:
*** Terminating app due to uncaught exception 'System.AggregateException', reason: 'System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> Couchbase.Lite.HttpResponseException: Exception of type 'Couchbase.Lite.HttpResponseException' was thrown.
  at Couchbase.Lite.Util.TransientErrorRetryHandler.HandleTransientErrors (System.Threading.Tasks.Task`1[TResult] request, System.Object state) <0x10146eae0 + 0x003ac> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at Couchbase.Lite.Util.TransientErrorRetryHandler+<ResendHandler>c__AnonStorey0.<>m__0 (System.Threading.Tasks.Task`1[TResult] t) <0x10146f150 + 0x0001f> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[TAntecedentResult,TResult].InnerInvoke () <0x100868880 + 0x00077> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task.Execute () <0x1008622d0 + 0x00043> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x1007e8670 + 0x00028> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x1007e7a50 + 0x000d3> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x1007e79a0 + 0x0008b> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x1007e7940 + 0x00047> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () <0x1007e8140 + 0x0001b> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Net.Http.HttpClient+<SendAsyncWorker>c__async0.MoveNext () <0x101184a60 + 0x00363> in <3afa4c02dd374bf7aa8b63dcf50f5aee#916f9e639ccb2e75c83a400c072f9687>:0 
   --- End of inner exception stack trace ---
  at Couchbase.Lite.CouchbaseLiteHttpClient+<SendAsync>c__AnonStorey2.<>m__1 (System.Threading.Tasks.Task`1[TResult] t) <0x101481f40 + 0x000b0> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[TAntecedentResult,TResult].InnerInvoke () <0x100868880 + 0x00077> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task.Execute () <0x1008622d0 + 0x00043> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
   --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) <0x100861980 + 0x00048> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) <0x10085d2b0 + 0x0006f> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task`1[TResult].get_Result () <0x10085d240 + 0x0002b> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at Couchbase.Lite.Internal.RemoteSession+<SendAsyncMultipartDownloaderRequest>c__AnonStorey2.<>m__0 (System.Threading.Tasks.Task`1[TResult] responseMessage) <0x101433e90 + 0x00947> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.ContinuationTaskFromResultTask`1[TAntecedentResult].InnerInvoke () <0x100868700 + 0x0006f> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task.Execute () <0x1008622d0 + 0x00043> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> Couchbase.Lite.HttpResponseException: Exception of type 'Couchbase.Lite.HttpResponseException' was thrown.
  at Couchbase.Lite.Util.TransientErrorRetryHandler.HandleTransientErrors (System.Threading.Tasks.Task`1[TResult] request, System.Object state) <0x10146eae0 + 0x003ac> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at Couchbase.Lite.Util.TransientErrorRetryHandler+<ResendHandler>c__AnonStorey0.<>m__0 (System.Threading.Tasks.Task`1[TResult] t) <0x10146f150 + 0x0001f> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[TAntecedentResult,TResult].InnerInvoke () <0x100868880 + 0x00077> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task.Execute () <0x1008622d0 + 0x00043> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x1007e8670 + 0x00028> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x1007e7a50 + 0x000d3> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x1007e79a0 + 0x0008b> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x1007e7940 + 0x00047> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () <0x1007e8140 + 0x0001b> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Net.Http.HttpClient+<SendAsyncWorker>c__async0.MoveNext () <0x101184a60 + 0x00363> in <3afa4c02dd374bf7aa8b63dcf50f5aee#916f9e639ccb2e75c83a400c072f9687>:0 
   --- End of inner exception stack trace ---
  at Couchbase.Lite.CouchbaseLiteHttpClient+<SendAsync>c__AnonStorey2.<>m__1 (System.Threading.Tasks.Task`1[TResult] t) <0x101481f40 + 0x000b0> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[TAntecedentResult,TResult].InnerInvoke () <0x100868880 + 0x00077> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task.Execute () <0x1008622d0 + 0x00043> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
   --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) <0x100861980 + 0x00048> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) <0x10085d2b0 + 0x0006f> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task`1[TResult].get_Result () <0x10085d240 + 0x0002b> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at Couchbase.Lite.Internal.RemoteSession+<SendAsyncMultipartDownloaderRequest>c__AnonStorey2.<>m__0 (System.Threading.Tasks.Task`1[TResult] responseMessage) <0x101433e90 + 0x00947> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.ContinuationTaskFromResultTask`1[TAntecedentResult].InnerInvoke () <0x100868700 + 0x0006f> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task.Execute () <0x1008622d0 + 0x00043> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
---> (Inner Exception #0) System.AggregateException: One or more errors occurred. ---> Couchbase.Lite.HttpResponseException: Exception of type 'Couchbase.Lite.HttpResponseException' was thrown.
  at Couchbase.Lite.Util.TransientErrorRetryHandler.HandleTransientErrors (System.Threading.Tasks.Task`1[TResult] request, System.Object state) <0x10146eae0 + 0x003ac> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at Couchbase.Lite.Util.TransientErrorRetryHandler+<ResendHandler>c__AnonStorey0.<>m__0 (System.Threading.Tasks.Task`1[TResult] t) <0x10146f150 + 0x0001f> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[TAntecedentResult,TResult].InnerInvoke () <0x100868880 + 0x00077> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task.Execute () <0x1008622d0 + 0x00043> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x1007e8670 + 0x00028> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x1007e7a50 + 0x000d3> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x1007e79a0 + 0x0008b> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x1007e7940 + 0x00047> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () <0x1007e8140 + 0x0001b> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Net.Http.HttpClient+<SendAsyncWorker>c__async0.MoveNext () <0x101184a60 + 0x00363> in <3afa4c02dd374bf7aa8b63dcf50f5aee#916f9e639ccb2e75c83a400c072f9687>:0 
   --- End of inner exception stack trace ---
  at Couchbase.Lite.CouchbaseLiteHttpClient+<SendAsync>c__AnonStorey2.<>m__1 (System.Threading.Tasks.Task`1[TResult] t) <0x101481f40 + 0x000b0> in <7dffa502a37a43879680038fad134195#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[TAntecedentResult,TResult].InnerInvoke () <0x100868880 + 0x00077> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
  at System.Threading.Tasks.Task.Execute () <0x1008622d0 + 0x00043> in <54e129cad3ee4c61a6b46c3680ddc69a#916f9e639ccb2e75c83a400c072f9687>:0 
---> (Inner Exception #0) [HttpResponseException: StatusCode = NotFound]<---
<---
<---

**Comment by borrrden** [comment]
Let me guess...this is Xamarin Android?  It seems to do this at will when exceptions are thrown but I've rarely seen it.  The actual exception looks like it might be related to another ticket that I am about to file, but it shouldn't be bringing down the entire process.  Why it does, and why it only seems to happen on Xamarin Android is a mystery.

**Comment by ykurniawan** [comment]
Hi,  Actually this log is from iOS app but Android also logs the same issue, and the  app also got terminated by this exception. Do you have an idea the right place to handle the exception? Thanks

**Comment by borrrden** [comment]
That's the thing...this exception is not meant to be handled by user code.  I still haven't quite figured out where or why this happens.  According to the message, every `Task` object must have its exception property checked or this will happen, but yet I can't reliably observe this even when provided reproduction cases and this path is one that checks it.  Maybe it's dependent on the version of Xamarin, I don't know.  For 2.0 the majority of this logic is being moved to C++ so this problem will become a non-issue but in terms of 1.x I'm having a hard time tracking down what to do about this.  I'm going to need more info from someone.

**Comment by borrrden** [comment]
Unfortunately 1.x has reached EOL without ever getting a reasonable fix for this.  If it is still an issue on 2.x then please open another issue.
